### PR TITLE
fix(filter-bar): enable sibling value dependent filters

### PR DESCRIPTION
**Pull request by HeartSquared**
## What

Enable Select/MultiSelect filters to load items based on a sibling's selected value.

- Selected values for dependent filters that are removed on items change are also removed from context values
- Found issues with the types, so fixed them

## Why

[KDS-1475](https://cultureamp.atlassian.net/jira/software/projects/KDS/boards/289?selectedIssue=KDS-1475)

Common use case for consumers.

[KDS-1475]: https://cultureamp.atlassian.net/browse/KDS-1475?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/rotten-icons-complain.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@kaizen/components": minor
+---
+
+- Update FilterBarMultiSelect to clear values when items change
```
Comment (gyfchong) [inline .changeset/rotten-icons-complain.md:5]: Is this more of a fix, cause it didn't work as expected before?

File: packages/components/src/FilterBar/subcomponents/FilterBarMultiSelect/FilterBarMultiSelect.tsx [hunk 2]
```diff
@@ -39,18 +41,37 @@ const convertConsumableFormatIntoSelection = (
...
   const filterState = getFilterState(id)
 
+  useEffect(() => {
+    if (!checkArraysMatch(items, propsItems)) {
+      setItems(propsItems)
+    }
+  }, [propsItems])
+
+  useEffect(() => {
+    if (Array.isArray(filterState.value)) {
```
[the remark this answers is not in the document]
Comment (mcwinter07) [reply packages/components/src/FilterBar/subcomponents/FilterBarMultiSelect/FilterBarMultiSelect.tsx:64]: in that they're not filtered out of the state if more than one is selected (for anyone reading this comment that doesn't understand our pairing mumbo jumbo) :P

File: packages/components/src/FilterBar/subcomponents/FilterBarMultiSelect/FilterBarMultiSelect.tsx [hunk 3]
```diff
@@ -39,18 +41,38 @@ const convertConsumableFormatIntoSelection = (
...
+      const itemValues = items.map(({ value }) => value)
+      const filteredValues = filterState.value.filter(value =>
+        itemValues.includes(value)
+      )
+
+      if (!checkArraysMatch(filterState.value, filteredValues)) {
```
Comment (mcwinter07) [inline packages/components/src/FilterBar/subcomponents/FilterBarMultiSelect/FilterBarMultiSelect.tsx:70]: Nice, no more engineers in the building! 😉

Translation: Nice, should be a less brittle way to update those values 🕺

**Review by mcwinter07 — APPROVED** [review]
Looks 🦖 🐜

**Pull request merged**